azerothcore/azerothcore-wotlk
# (Core/Quest) Refresh SMSG_QUESTGIVER_STATUS_MULTIPLE after a quest change

This updates the status of nearby quest givers and prevents them from being stuck in a state that's no longer valid (for example, stuck with yellow question mark until area update).
This issue was discovered and was made in support of Malcrom's Quest Scripting https://github.com/azerothcore/azerothcore-wotlk/pull/8895 where it became even more fraustrating.

This will force the quests status to update while being in the same area without the need to relog or re-enter the area id.

## Changes Proposed:
-  Move SendQuestGiverStatusMultiple
-  Create SendQuestGiverStatusMultiple

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/8800
- Supports PR https://github.com/azerothcore/azerothcore-wotlk/pull/8895 and various other quests.

## Tests Performed:
Builds and compiles with no issues. Tested and seems to work for me now.

## How to Test the Changes:

1. Best and quickest way to perform this is to find a quest giver that is both a start and turn in of a quest.
2. Take the quest from the quest giver natuarlly
3. Do not leave 
4. Do a .quest complete of the said quest, notice
5. You will notice the ! will go from yellow to grey ? on acceptance, and will then turn yellow when quest complete is done.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
